pyproject.toml

```diff
@@ -26,28 +26,33 @@ documentation = "https://stfc.github.io/janus-core/"
 janus = "janus_core.cli.janus:app"
 
 [tool.poetry.dependencies]
-python = "^3.9"
 ase = "^3.23"
-chgnet = "0.3.8"
-dgl = "2.1.0" # Pin due to matgl installation issues
+codecarbon = "^2.5.0"
 mace-torch = "0.3.6"
-matgl = "1.1.2"
 numpy = "^1.26.4"
-pyyaml = "^6.0.1"
-typer = "^0.9.0"
-typer-config = "^1.4.0"
 phonopy = "^2.23.1"
+python = "^3.9"
+pyyaml = "^6.0.1"
 seekpath = "^1.9.7"
 spglib = "^2.3.0"
-torchdata = "0.7.1" # Pin due to dgl issue
+torch = ">= 2.1, <= 2.2" # Range required for dgl
 torch-dftd = "0.4.0"
-codecarbon = "^2.5.0"
+typer = "^0.9.0"
+typer-config = "^1.4.0"
+
 alignn = { version = "2024.5.27", optional = true }
+chgnet = {version = "0.3.8", optional = true}
+dgl = { version = "2.1.0", optional = true } # Pin due to matgl installation issues
+matgl = { version = "1.1.3", optional = true}
 sevenn = { version = "0.9.3", optional = true }
+torchdata = {version = "0.7.1", optional = true} # Pin due to dgl issue
 torch_geometric = { version = "^2.5.3", optional = true }
 
 [tool.poetry.extras]
+all = ["alignn", "chgnet", "matgl", "dgl", "torchdata", "sevenn", "torch_geometric"]
 alignnff = ["alignn"]
```

Comment:
Do we want to change this to just "alignn", or change the calculator architecture name to match this, so we're consistent , @alinelena?

Reply:
yes fine to change it to alignn... why do we have torchdata, geometris and dgl as options... I would expect them to be deps of specific targets, sevenn and matgl/chgnet?

Reply:
Cool ok

Those other dependencies aren't actual options you'd select, the extra options are `all`, `alignn`, `chgnet`, `m3gnet` and `sevennet`, the others are just dependencies within those that we have to specify to fix installation issues (either because they're missing, or we need to pin them)

all = ["alignn", "chgnet", "matgl", "dgl", "torchdata", "sevenn", "torch_geometric"]
alignn = ["alignn"]
chgnet = ["chgnet"]
m3gnet = ["matgl", "dgl", "torchdata"]
sevennet = ["sevenn", "torch_geometric"]